### ISSUE TITLE
feat: validate relay messages using timestamp

### ISSUE THREAD

**Issue by alrevuelta** [waku-org/nwaku#1542]
### Problem
Few days ago I did some tests in waku network, sending multiple messages per second of a size close to 1 Mb. The result was that memory consumption of the nodes really increased and after stopping sending messages, the network kept gossiping them for more than 1 hour. In other words, after stopping, some messages were received more than 1h later.

I suspect this happened because messages were queued in the node (explaining mem increase) and since they were i) a lot of them and ii) quite big in size, it took long time to empty it, which can explain this long delay.

So, would like to raise the question, if we should do some validation of the messages per timestamp. Meaning that if a message contains a timestamp that differs more than `x` from the current timestamp of the node, then we drop it and stop relaying it. This "incentivices" nodes to use accurante timestamps and we avoid having echoes from hour ago.

In the meantime, I will rerun this experiment in a private setup, and will try to extract more concrete findings, to be added to this issue. Need to check also with nim-libp2p team, to check if there are some measurements on this side to prevent this "echoes" from happening.

### Suggested solution
TBD

### Acceptance criteria
TBD

**Comment by jm-clius** [comment]
Note that currently `timestamp` is optional and can easily be fudged. Generally changing the requirements for a `WakuMessage` to have a synchronised, mandatory timestamp, will be a fairly significant shift in RFC (and requirements on devices using Waku). I can imagine that for certain pubsubTopics and applications such validation can be enabled, but then this may venture into the realm of App-Level validation

**Comment by alrevuelta** [comment]
Interesting I see. Will then change the title to something that explains the problem, leaving the solution open. I think we agree that we don't want messages being gossiped one hour later right? These messages will take bandwidth but are no longer valid? Or perhaps they are? Note that I still have to do more anlysis on what happened in this experiment I run.

Regarding using the timestamp, understand its not a trivial change, specifically since its not mandatory. Regarding "requirements on devices using Waku" which requirements do you mean? Making the timestamp mandatory is the requirement change? Or is there anyting beyond that?

tldr: will focus on the problem instead of the solution.

**Comment by jm-clius** [comment]
Could be related to: https://github.com/waku-org/nwaku/issues/1662
